universal/include/userver/formats/parse/try_parse.hpp

```diff
@@ -28,25 +28,33 @@ inline bool Is(Value&& value) {
   }
 }
 
-inline constexpr utils::TypeList<bool, int, std::int64_t, std::uint64_t, double, std::string> kBaseTypes;
+inline constexpr utils::impl::TypeList<bool, std::int64_t, std::uint64_t, double, std::string> kBaseTypes;
 
 
 } // namespace impl
 
 
 template <typename T, typename Value>
-inline std::enable_if_t<utils::AnyOf(utils::IsSameCarried<T>(), impl::kBaseTypes), std::optional<T>>
+constexpr inline std::enable_if_t<utils::impl::AnyOf(utils::impl::IsConvertableCarried<T>(), impl::kBaseTypes), std::optional<T>>
 TryParse(Value&& value, userver::formats::parse::To<T>) {
-  if(value.IsMissing() || value.IsNull() || !impl::Is<T>(value)) {
+  if(!impl::Is<T>(value)) {
     return std::nullopt;
   }
-  return value.template As<T>();
+  auto obj = value.template As<std::optional<T>>();
```

Comment:
The `As<T>` was correct here

If missing or null, then the above will `return std::nullopt`. You can leave a comment about it for future generations if it may be confusing

Otherwise we know that `value` contains exactly our desired type, we should parse it and wrap in an `optional`. It's simpler logic and more performant than to parse `optional` here